common/flatpak-context.c

```diff
@@ -3162,6 +3162,10 @@ flatpak_context_get_allowed_exports (FlatpakContext *context,
     {
       g_ptr_array_add (allowed_extensions, g_strdup (".ini"));
     }
+  else if (strcmp (source_path, "share/krunner/dbusplugins") == 0)
+    {
+      g_ptr_array_add (allowed_extensions, g_strdup (".desktop"));
```

Comment:
Are these sufficiently similar to ordinary .desktop files that we can be confident that KDE won't add something like

```
[Some other group]
ThisWillExecuteArbitraryCodeToo=sh -c 'echo pwned > /tmp/this-could-have-been-malicious'
```

and accidentally allow apps to bypass Flatpak's filtering of the `Exec` field?

Reply:
I'm not 100% sure what your question aims at, but the krunner sources only read the **Desktop Entry** group. I can't find anything that does an "exec" in krunner/dbusplugins located desktop files.

Reply:
What I'm trying to establish is whether this could become a security vulnerability later.

If the only thing krunner does with this is to either communicate with the D-Bus name or execute a program from the `Exec` field, then this isn't creating a new security vulnerability, because communicating with the D-Bus name is safe, and Flatpak filters the contents of the `Exec` field to turn it into a suitable, safe `flatpak run` command.

But, if there is a chance that the maintainers of krunner might add a new field like the one I described in some future version and run its contents automatically in some situation, then that would create a new security vulnerability, because apps would be able to specify whatever content they want for that field (Flatpak would not be filtering it).

Does that make sense?

If you're a krunner maintainer or if you're in contact with the krunner maintainers, the way to address this concern would be for the krunner maintainers to tell us: yes we're aware of that as a potential cause of security vulnerabilities, and we'll avoid creating that situation.

Reply:
krunner just uses DBus there, so this is basically no problem at all. But I'll try to reach out for the krunner devs to get a comment here...

Reply:
>But, if there is a chance that the maintainers of krunner might add a new field like the one I described in some future version and run its contents automatically in some situation

This is for sure not the case. We only need the runners to respond to DBus calls.